lib/pbench/agent/tool_meister.py

```diff
@@ -1534,37 +1540,34 @@ def end_tools(self, data):
                 )
                 failures += 1
 
-        # Remove persistent tool temporary working directory on remote host
+        # Remove persistent tool temporary working directory
         directory = data["directory"]
         tool_dir = self.directories[directory]
 
-        if self._hostname == self._controller:
-            directory_to_delete = tool_dir
-        else:
-            directory_to_delete = tool_dir.parent
-
-        self.logger.info(
+        self.logger.debug(
             "%s: deleting persistent tool tmp directory %s %s",
             self._hostname,
             self._group,
-            directory_to_delete,
+            tool_dir,
         )
-        unexpected_files = [x for x in directory_to_delete.iterdir() if x.is_file()]
-        if not unexpected_files:
-            try:
-                shutil.rmtree(directory_to_delete)
-            except Exception:
-                self.logger.exception(
-                    f"Failed to remove tmp persistent tool data hierarchy, "
-                    f"{directory_to_delete}"
-                )
-                failures += 1
-        else:
+        unexpected_files = []
+        for dir, _, files in os.walk(tool_dir):
+            dirpath = Path(dir).relative_to(tool_dir)
+            if files:
+                unexpected_files += map(lambda x: f"{dirpath}/{x}", files)
+
+        if unexpected_files:
+            self.logger.warning(
+                "%s: unexpected temp files %s",
+                self._hostname,
+                ",".join(unexpected_files),
+            )
```

Comment:
I would use the following (but, I'm just **that** anal-retentive 😆):
```
        if unexpected_files:
            self.logger.warning(
                "%s: removing unexpected temp file%s: %s",
                self._hostname,
                "s" if len(unexpected_files) > 1 else "",
                ",".join(unexpected_files),
            )
```
(That is, pluralize `file` only if there is more than one file, and add a colon after it; also, add "removing".)

Reply:
I liked the suggestion but this case almost will never happen (at least we hope so) and if there are number of files getting written under this directory then either we have a bug somewhere or some other program is using the directory created by us. So I don't want to make change here after I got all the approvals :stuck_out_tongue_closed_eyes: